.buildkite/e2e/reporter/main.go

```diff
@@ -0,0 +1,145 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/joshdk/go-junit"
+)
+
+var (
+	// extractSlugNameRe is a regexp to extract the name of the test environment.
+	// It is set by the pipeline generator via EnvVarClusterName.
+	extractSlugNameRe = regexp.MustCompile("e2e-tests-eck-e2e-(.*)-[a-z]*-[0-9]*.xml")
+)
+
+func main() {
+	if len(os.Args) != 2 {
+		exitWith(errors.New("argument 'directory' required"))
+	}
+
+	failures := 0
+
+	// process all xml report in the given diretory
+	err := filepath.Walk(os.Args[1], func(xmlReportPath string, info os.FileInfo, err error) error {
+		if err != nil {
+			exitWith(err)
+		}
+		if info.IsDir() {
+			return nil
+		}
+		if !strings.HasSuffix(xmlReportPath, ".xml") {
+			return nil
+		}
+
+		// extract slug name from the report file name
+		match := extractSlugNameRe.FindStringSubmatch(xmlReportPath)
+		if len(match) != 2 {
+			return fmt.Errorf("failed to extract slug name from %s", xmlReportPath)
+		}
+		slugName := match[1]
+
+		// ingest xml report to work with structs
+		xmlReport, err := os.ReadFile(xmlReportPath)
+		if err != nil {
+			return err
+		}
+		// a suite groups all the tests of a package
+		suites, err := junit.Ingest(xmlReport)
+		if err != nil {
+			return err
+		}
+
+		failedTests, passedTests := sortTests(suites)
+
+		failures += printFinalReport(slugName, failedTests, passedTests)
+
+		return nil
+	})
+	if err != nil {
+		exitWith(err)
+	}
+
+	if failures > 0 {
+		os.Exit(1)
+	}
+}
+
+func sortTests(suites []junit.Suite) ([]junit.Test, []junit.Test) {
+	failedTests := []junit.Test{}
+	passedTests := []junit.Test{}
+	failedTestsMap := map[string]junit.Test{}
+
+	// traverse all suites to find failed and passed tests
+	for _, suite := range suites {
+		if len(suite.Tests) == 0 {
+			continue
+		}
```

Comment:
Is the above check necessary, since the below for loop will just do nothing?

Reply:
It can be removed.